clang/lib/CodeGen/Targets/PPC.cpp

```diff
@@ -337,12 +350,77 @@ CharUnits PPC32_SVR4_ABIInfo::getParamTypeAlignment(QualType Ty) const {
   return CharUnits::fromQuantity(4);
 }
 
+ABIArgInfo PPC32_SVR4_ABIInfo::handleComplex(QualType Ty,
+                                             uint64_t &TypeSize) const {
+
+  assert(Ty->isAnyComplexType());
```

Comment:
It would also be good if the assert has a message.
Additionally, for people who do not do assert builds, since `Ty` is only used for the assert, they may get an unused variable warning.

Reply:
if use only for the assert, suggest putting assert outside the function,